sdk/core/AzureCore/Source/Pipeline/Transport/HTTPRequest.swift

```diff
@@ -38,16 +41,22 @@ public class HttpRequest: HttpMessage {
         return comps
     }
 
-    public init(httpMethod: HttpMethod, url: String, headers: HttpHeaders, files: [String]? = nil,
-                data: Data? = nil) {
-        self.httpMethod = httpMethod
+    // MARK: Initializers
+
+    public init(method: HTTPMethod, url: String,
+                queryParams: [String: String], headerParams: HTTPHeaders,
+                files: [String]? = nil, data: Data? = nil) {
+        self.httpMethod = method
         self.url = url
-        self.headers = headers
+        self.headers = headerParams
```

Comment:
I think I prefer the old name `headers` - `queryParams` has the word "Params" in it only because we're allowing the caller to specify "query-string parameters" and "headers".

Reply:
Yep, makes sense.